client/post-editor/editor-drawer/index.jsx

```diff
@@ -326,7 +327,10 @@ const EditorDrawer = React.createClass( {
 				{ site && ! this.hasHardCodedPostTypeSupports( type ) && (
 					<QueryPostTypes siteId={ site.ID } />
 				) }
-				{ this.renderTaxonomies() }
+				{ config.isEnabled( 'manage/custom-post-types' ) &&	! includes( [ 'post', 'page' ], this.props.type )
```

Comment:
Makes sense, I haven't moved taxonomies to be async-loaded because it was causing issues. Leaving that up to your or @timmyc's discretion :)